src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForSchemaReadTests.cs

```diff
@@ -0,0 +1,410 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.UnitTests;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+
+namespace Desktop.Analyzers.UnitTests
+{
+    public partial class DoNotUseInsecureDTDProcessingAnalyzerTests : DiagnosticAnalyzerTestBase
+    {
+        private DiagnosticResult GetCA3075CSharpResultAt(int line, int column, string name)
+        {
+            return GetCSharpResultAt(line, column, CA3075RuleId, string.Format(CA3075LoadXmlMessage, name, "Read"));
+        }
+
+        private DiagnosticResult GetCA3075BasicResultAt(int line, int column, string name)
+        {
+            return GetBasicResultAt(line, column, CA3075RuleId, string.Format(CA3075LoadXmlMessage, name, "Read"));
+        }
+
+        [Fact]
+        public void UseXmlSchemaReadShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Xml;
+using System.Xml.Schema;
+
+namespace FxCopUnsafeXml
+{
+    public class UseXmlReaderForSchemaRead
```

Comment:
If the name of this test class is referring to the type of the first argument to `XmlSchema.Read`, it should be named `UseTextReaderForSchemaRead`.